recipes/PeoplesSpeech/peoples_speech_prepare.py

```diff
@@ -0,0 +1,398 @@
+    audio files will NOT be extracted but instead read from shard directly. Instead
+    we propose to still generate .csv files to have transcriptions and durations
+    readable to the user.
+
+    Download: https://huggingface.co/datasets/MLCommons/peoples_speech
+    Reference: https://arxiv.org/abs/2111.09344
+
+    The `train.csv` file is created by combining the sets given in the `subsets`
+    variable.
+
+    The `dev.csv` and `test.csv` files are created based on the `DEV` and `TEST` splits
+    specified in the `splits` list.
+
+    Parameters
+    ----------
+    hf_download_folder : str
+        The path where HF stored the dataset. Important, you must set the global
+        env variable HF_HUB_CACHE to the same path as HuggingFace is primilarily
+        using this to know where to store datasets.
+    save_folder : str
+        The path to the folder where the CSV files will be saved.
+    subsets : list
+        Target subset. People's speech contains multiple subsets, which must be
+        loaded invidividually and then concatenated. E.g. 'clean', 'clean_sac',
+        'dirty' or 'dirty_sa'. E.g. to combine  ['clean', 'dirty'].
+    skip_prep : bool, optional
+        If True, the data preparation will be skipped, and the function will return immediately.
+
+    Returns
+    -------
+    None
+    """
+
+    if not os.path.isfile(hf_download_folder):
+        msg = "You must download the dataset with HuggingFace before starting "
+        msg += (
+            "this recipe. Please check the HuggingFace hub of people's speech."
+        )
+        raise ValueError(msg)
+
+    if skip_prep:
+        logger.info("Skipping data preparation as `skip_prep` is set to `True`")
+        return
+
+    if len(subsets) == 0:
+        raise ImportError(
+            "At least one People's speech subset must be specified."
+        )
+
+    # Setting output paths
+    save_output = {}
+    splits = ["train", "validation", "test"]
+    for split in splits:
+        save_output[split] = os.path.join(save_folder, str(split) + ".csv")
+
+    # check if the data is already prepared
+    if skip_csv(save_output):
+        logger.info("Skipping preparation, completed in previous run.")
+        return
+    else:
+        logger.info("Starting data preparation...")
+
+    hf_dataset = load_and_concatenate_datasets(subsets, hf_download_folder)
+
+    logger.info(
+        f"Preparing CSV of the Peoples Speech dataset in {save_folder}..."
+    )
+
+    os.makedirs(save_folder, exist_ok=True)
+
+    i = 0
+    for split, output in save_output.items():
+        logger.info(f"Starting creating {output} using {split} split.")
+        HF_create_csv(output, hf_dataset[i], split)
+        i += 1
+
+    logger.info("Data preparation completed!")
+
+
+def load_and_concatenate_datasets(subsets, hf_download_folder):
+    """Load/download and concatenate all the specified subsets from People's
+    speech
+    """
```

Comment:
missing docstring for args

Reply:
i think function is lacking a lot of details of what is happening here e.g. caching etc